aptos-move/replay-benchmark/src/overrides.rs

```diff
@@ -73,6 +125,114 @@ impl OverrideConfig {
             state_override.insert(gas_schedule_state_key, gas_schedule_state_value);
         }
 
+        // Override packages.
+        let mut overridden_package_registries = HashMap::new();
+        for package in &self.package_override.packages {
+            // Modify existing package metadata or add new one.
+            let package_address = package
+                .modules()
+                .map(|m| m.self_addr())
+                .last()
+                .expect("Package must contain at least one module");
+            let package_registry_state_key =
+                StateKey::resource(package_address, &PackageRegistry::struct_tag())
+                    .expect("Should always be able to create state key for package registry");
+
+            let old_package_state_value =
+                match overridden_package_registries.remove(&package_registry_state_key) {
+                    Some(state_value) => state_value,
+                    None => state_view
+                        .get_state_value(&package_registry_state_key)
+                        .unwrap_or_else(|err| {
+                            panic!(
+                                "Failed to fetch package registry at {}: {:?}",
+                                package_address, err
+                            )
+                        })
+                        .expect("Package registry for override must always exist"),
+                };
+
+            let metadata = package.extract_metadata().unwrap_or_else(|err| {
+                panic!(
+                    "Failed to extract metadata for package {}: {:?}",
+                    package.name(),
+                    err
+                )
+            });
+            let new_package_state_value = old_package_state_value
+                .map_bytes(|bytes| {
+                    let mut package_registry = bcs::from_bytes::<PackageRegistry>(&bytes)
+                        .expect("Package registry should deserialize");
+
+                    let mut metadata_idx = None;
+                    for (idx, package_metadata) in package_registry.packages.iter().enumerate() {
+                        if package_metadata.name == metadata.name {
+                            metadata_idx = Some(idx);
```

Comment:
we can assign it here and only handle None below?

Reply:
I am not sure, you need to clone metadata then (Rust complains)? Even though it is clearly that metadata cannot be moved if we fallback to break.